src/plugins/data/common/search/aggs/agg_configs.ts

```diff
@@ -155,11 +155,16 @@ export class AggConfigs {
       return agg.enabled;
     };
 
-    const aggConfigs = new AggConfigs(this.indexPattern, this.aggs.filter(filterAggs), {
-      typesRegistry: this.typesRegistry,
-    });
-
-    return aggConfigs;
+    return new AggConfigs(
+      this.indexPattern,
+      this.aggs.filter(filterAggs),
+      {
+        ...this.opts,
```

Comment:
@ppisljar we expect that `hierarchical` and  `partialRows` properties to be excluded when cloned or not?, looks like a bug

I see in `esaggs.ts` we mutate that object.

<img width="663" alt="image" src="https://user-images.githubusercontent.com/20072247/171905425-383b3d6f-9ef4-486d-addb-7a16fec60fcb.png">

This means it will be lost when cloned.